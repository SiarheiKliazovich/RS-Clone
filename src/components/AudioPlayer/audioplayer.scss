@import "../../sass/variables";
@import "../../sass/mixins";



.voice-btn-fixed {
  position: fixed;
  bottom: 40px;
  left: 40px;
  margin-bottom: 60px;
  border: none;
  color: #fff;
  background: transparent;
  z-index: 999;
  transition: all .3s ease-in-out;
  cursor: pointer;
  overflow: visible;

    &::before {
      content: "";
      position: absolute;
      z-index: 0;
      left: 50%;
      top: 50%;
      transform: translateX(-50%) translateY(-50%);
      display: block;
      background: rgba(209,25,25,.3);
      border-radius: 50%;
      animation: mb 1.5s ease-out infinite;
      width: 70px;
      height: 70px;
    }

    &::after {
    content: "";
    position: absolute;
    z-index: 0;
    left: 50%;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
    display: block;
    background: rgba(209,25,25,.3);
    border-radius: 50%;
    animation: mb 1.5s ease-out infinite;
    width: 50px;
    height: 50px;
    }
}


.voice-btn-fixed .voice-btn {
  width: 45px;
  height: 45px;
  line-height: 50px;
  background: #ff4b4b;
  -ms-box-shadow: 0 2px 4px rgba(0,0,0,.07);
  -o-box-shadow: 0 2px 4px rgba(0,0,0,.07);
  box-shadow: 0 2px 4px rgb(0 0 0 / 7%);
  border-radius: 50%;
  z-index: 9;
  cursor: pointer;
}

// .voice-btn-fixed i {
//   font-size: 18px;
//   transition: all .3s ease-in-out;
// }



.volume-up {
  background: url(../../assets/images/png/on.png);
  font-weight: 900;
  font-style: normal;
  font-variant: normal;
  text-rendering: auto;
  line-height: 1;
  font-size: 18px;
  display: inline;
  transition: all .3s ease-in-out;
  width: 18px;
  height: 18px;
  background-repeat: no-repeat;
}



.volume-mute {
  font-weight: 900;
  display: inline;
  font-style: normal;
  font-variant: normal;
  text-rendering: auto;
  line-height: 1;
  font-size: 18px;
  width: 18px;
  height: 18px;
  transition: all .3s ease-in-out;
  background:  url(../../assets/images/png/off.png);
  background-repeat: no-repeat;
   
  }

.none {
  display: none;
}
