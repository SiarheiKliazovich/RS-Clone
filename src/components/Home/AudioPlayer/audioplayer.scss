@import "../../../sass/variables";
@import "../../../sass/mixins";

.voice-btn-fixed {
  position: fixed;
  bottom: 40px;
  left: 40px;
  margin-bottom: 60px;
  border: none;
  color: #fff;
  background: transparent;
  z-index: 999;
  transition: all .3s ease-in-out;
  cursor: pointer;
  overflow: visible;

  &::before {
    content: "";
    position: absolute;
    z-index: 0;
    left: 50%;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
    display: block;
    background: rgba(209,25,25,.3);
    border-radius: 50%;
    animation: mb 1.5s ease-out infinite;
    width: 70px;
    height: 70px;
  }

  &::after {
    content: "";
    position: absolute;
    z-index: 0;
    left: 50%;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
    display: block;
    background: rgba(209,25,25,.3);
    border-radius: 50%;
    animation: mb 1.5s ease-out infinite;
    width: 50px;
    height: 50px;
  }
}

.voice-btn-fixed .voice-btn {
  width: 45px;
  height: 45px;
  line-height: 50px;
  background: #ff4b4b;
  -ms-box-shadow: 0 2px 4px rgba(0,0,0,.07);
  -o-box-shadow: 0 2px 4px rgba(0,0,0,.07);
  box-shadow: 0 2px 4px rgb(0 0 0 / 7%);
  border-radius: 50%;
  z-index: 9;
  cursor: pointer;

  &:hover {
    background: #fff;
  }
}

.volume-up {
  background: url(../../../assets/images/png/on.png);
  transition: all .3s ease-in-out;
  width: 18px;
  height: 18px;
  background-repeat: no-repeat;
}

.volume-mute {
  width: 18px;
  height: 18px;
  transition: all .3s ease-in-out;
  background:  url(../../../assets/images/png/off.png);
  background-repeat: no-repeat;
}

.none {
  display: none;
}

@keyframes mb {
  0% {
      transform: translateX(-50%) translateY(-50%) translateZ(0) scale(1);
      -webkit-opacity: 1;
      -moz-opacity: 1;
      -ms-opacity: 1;
      -o-opacity: 1;
      opacity: 1;
  }
  to {
      transform: translateX(-50%) translateY(-50%) translateZ(0) scale(1.5);
      -webkit-opacity: 0;
      -moz-opacity: 0;
      -ms-opacity: 0;
      -o-opacity: 0;
      opacity: 0;
  }
}
