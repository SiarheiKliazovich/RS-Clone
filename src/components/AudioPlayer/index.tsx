import React, { useState } from "react";
import useSound from "use-sound";
import classNames from "classnames";
import soundUrl from "../../assets/mp3/Eric Clapton & B.B.King - Three O`Clock BLues.mp3";
import "./audioplayer.scss";

const AudioPlayer = () => {
  const [play, { stop }] = useSound(soundUrl, { volume: 0.5 });
  const [on, setOn] = useState(false);

  const handleClick = () => {
    setOn(true);
    play();
  };
  const Click = () => {
    setOn(false);
    stop();
  };

  return (
    <button
      type="button"
      className="voice-btn-fixed"
      id="my_audio"
      onClick={() => handleClick()}
      onMouseEnter={() => Click()}
    >
      <div className="voice-btn">
        <img className={classNames("volume-up", { none: on === false })} />
        <img className={classNames("volume-mute", { none: on === true })} />
      </div>
    </button>
  );
};

export default AudioPlayer;
