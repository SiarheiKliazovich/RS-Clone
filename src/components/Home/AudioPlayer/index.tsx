import { useState } from "react";
import useSound from "use-sound";
import classNames from "classnames";
import soundUrl from "../../../assets/mp3/Eric Clapton & B.B.King - Three O`Clock BLues.mp3";
import "./audioplayer.scss";

const AudioPlayer = () => {
  const [play, { stop }] = useSound(soundUrl, { volume: 0.3 });
  const [on, setOn] = useState(false);

  const hand = (on: boolean) => {
    if (on) {
      stop();
      setOn(false);
    } else {
      play();
      setOn(true);
    }
  };

  return (
    <button
      type="button"
      className="voice-btn-fixed"
      id="my_audio"
      onClick={() => hand(on)}
    >
      <div className="voice-btn">
        <div className={classNames("volume-up", { none: on === false })}></div>
        <div className={classNames("volume-mute", { none: on === true })}></div>
      </div>
    </button>
  );
};

export default AudioPlayer;
