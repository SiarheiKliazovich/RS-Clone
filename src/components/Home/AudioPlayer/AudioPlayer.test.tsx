import { render } from "@testing-library/react";
import AudioPlayer from ".";

describe("AudioPlayer", () => {
  it("renders correctly", () => {
    const { container } = render(<AudioPlayer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
