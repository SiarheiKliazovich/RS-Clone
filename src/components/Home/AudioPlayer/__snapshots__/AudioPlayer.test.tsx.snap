// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer renders correctly 1`] = `
<button
  class="voice-btn-fixed"
  id="my_audio"
  type="button"
>
  <div
    class="voice-btn"
  >
    <div
      class="volume-up none"
    />
    <div
      class="volume-mute"
    />
  </div>
</button>
`;
